securedrop-admin

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python2
+"""
+SecureDrop Admin Toolkit.
+
+For use by administrators to install, maintain, and manage their SD
+instances.
+"""
+
+import argparse
+import subprocess
+import os
+import sys
+import getpass
+
+
+SD_DIR = os.path.dirname(os.path.realpath(__file__))
+ANSIBLE_PATH = os.path.join(SD_DIR, "./install_files/ansible-base")
+SITE_CONFIG = os.path.join(ANSIBLE_PATH, "group_vars/all/site-specific")
+VENV_DIR = os.path.join(SD_DIR, ".venv")
+VENV_ACTIVATION = os.path.join(VENV_DIR, 'bin/activate_this.py')
+
+
+def error(msg, level='warning', exit=True):
+    """ Stupid wrapper function to normalize warnings """
+    print("{} - {}".format(level.upper(), msg))
+    if exit:
+        sys.exit(1)
+
+
+def activate_venv():
+    """Use to activate the local virtualenv"""
+    try:
+        execfile(VENV_ACTIVATION, dict(__file__=VENV_ACTIVATION))
+    except IOError:
+        error("Pre-requisites not in place, re-run command with 'setup'",
+              level='error')
+
+
+def sdconfig(args):
+    """Configure SD site settings"""
+    activate_venv()
+    subprocess.call(['./configure'], cwd=ANSIBLE_PATH, env=os.environ.copy())
+
+
+def envsetup(args):
+    """Install SD pre-reqs"""
+    # virtualenv doesnt exist? Install dependencies and create
+    if not os.path.exists(VENV_ACTIVATION):
+        subprocess.call(['sudo', 'su', '-c',
+                        "apt-get update && \
+                        apt-get install -y \
+                        python-virtualenv \
+                        python-pip \
+                        ccontrol \
+                        virtualenv \
+                        libffi-dev \
+                        libssl-dev \
+                        libpython2.7-dev"])
+
+        # Technically you can create a virtualenv from within python
+        # but pip can only be run over tor on tails, and debugging that
+        # along with instaling a third-party dependency is not worth
+        # the effort here.
+        subprocess.call(['torify', 'virtualenv', VENV_DIR])
+
+    pip_install_cmd = [
+            'torify',
+            os.path.join(VENV_DIR, 'bin', 'pip'),
+            'install',
+            ]
+    requirements_cmd = ['-r', os.path.join(ANSIBLE_PATH, 'requirements.txt'),
+                        '--require-hashes']
+
+    if args.update:
+        subprocess.call(pip_install_cmd+['-U']+requirements_cmd)
+    else:
+        subprocess.call(pip_install_cmd+requirements_cmd)
+
+
+def install_securedrop(args):
+    """Install/Update SecureDrop"""
+    activate_venv()
+    # Yaml library cannot be imported until virtualenv is activated
+    # (hence the yaml library is not imported up-top)
+    import yaml
+
+    try:
+        site_config_vars = yaml.safe_load(open(SITE_CONFIG, 'ro').read())
```

Comment:
@fowlslegs is correct that you should explicitly close the file, but since you're just reading the contents all in one go and then never using the file handle again, IMHO it would be cleaner to use a context manager (which will automatically handle closing the file for you). I would also encourage you to keep commands and exceptions as close together as possible, so it's easier to reason about your code. Something like this:

```py
try:
    with open(SITE_CONFIG) as site_config_file:
        site_config = yaml.safe_load(site_config_file.read())
except IOError:
    # Handle IOError
except yaml.YAMLError:
    # Optional: gracefully handled malformed YAML. Might not be necessary
else:
    # Nothing went wrong! Do stuff with site_config.
```